buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -389,6 +389,81 @@ private static long compareUintBigEndianB(
         return 0;
     }
 
+    private static final class SWARByteSearch {
+
+        private static final long[] SEARCH_PATTERNS;
+
+        static {
+            SEARCH_PATTERNS = new long[256];
+            for (int i = 0; i < SEARCH_PATTERNS.length; ++i) {
+                final byte b = (byte) (i - 128);
+                SEARCH_PATTERNS[i] = compilePattern(b);
```

Comment:
This is 2KB, but if you change this to the multiplication mentioned earlier, I don't see benefit from caching

Reply:
That's better, it's now very cheap, great